salt/modules/power_dns.py

```diff
@@ -0,0 +1,567 @@
+"""
+Create, modify and delete PowerDNS zone
+"""
+
+import logging
+import socket
+
+import requests
+from salt.exceptions import CommandExecutionError
+
+# Constant section
+
+DEFAULT_TTL = 300  # In Seconds
+DEFAULT_NS_TTL = 3600  # In Seconds
+
+REPLACE_CHANGETYPE = "REPLACE"
+DELETE_CHANGETYPE = "DELETE"
+
+DNS_RECORD_NS = "NS"
+DNS_RECORD_SOA = "SOA"
+DNS_RECORD_TXT = "TXT"
+DNS_RECORD_A = "A"
+DNS_RECORD_CNAME = "CNAME"
+
```

Comment:
Personally I woulf add `PTR`, `MX`, `CNAME` and `AAAA` record types as well

Reply:
Module/state can work with PTR, MX, CNAME, and AAAA records types as well as with TXT, A, and CNAME. These constants are declared here cause there are additional conditionals for these records in the module/state code.